### PR TITLE
Reduce file size of mogan draw on wasm

### PR DESCRIPTION
**Pull request by Oyyko**
## Why you open this Pull Request?

Reduce the file size of mogan draw on wasm

## What work have you done in the current Pull Request?

- [x] remove the unnecessary plugins on wasm.

### CODE REVIEW

File: xmake/draw.lua [hunk 1]
```diff
@@ -63,13 +63,17 @@ function add_target_draw()
...
+
+    -- reduce the file size on wasm
+    if not is_plat("wasm") then
+      add_files(plugin_freetype_srcs)
+      add_files(plugin_database_srcs)
+      add_files(plugin_metafont_srcs)
+      add_files(plugin_openssl_srcs)
+      add_files(plugin_xml_srcs)
+      add_files(plugin_ghostscript_srcs)
+      add_files("src/Mogan/Draw/draw.cpp")
```
Comment (darcy-shen) [inline xmake/draw.lua:75]: this entry is necessary for wasm

Comment (Oyyko) [reply xmake/draw.lua:75]: The CI even passed. Maybe something wrong with CI? I will fix this now.

Comment (darcy-shen) [reply xmake/draw.lua]: not wrong, because the entry is the top level one.

**Comment by darcy-shen** [comment]
Trigger the deploy workflow manually: https://github.com/XmacsLabs/mogan/actions/workflows/cd_draw_mogan_app.yml

Click the `Run workflow` button.

**Comment by darcy-shen** [comment]
`Build and test on wasm` is for Mogan Research on wasm, we haven't set ci for Mogan Draw.

**Comment by darcy-shen** [comment]
According to the manually triggered ci, it failed to work.

**Comment by Oyyko** [comment]
It seems that the `draw.cpp` still uses some function that were designed in `plugin`. So can I delete these code and see whether draw app can still run without these plugin codes?

**Comment by darcy-shen** [comment]
Reducing the size of draw.data is much easier than draw.wasm, I suggest that you should work on reducing draw.data.

**Comment by darcy-shen** [comment]
If you want to reduce the size of draw.wasm, please follow the tasks in project 48:
+ https://github.com/XmacsLabs/mogan/pull/1148
+ https://github.com/XmacsLabs/mogan/pull/1099

**Comment by darcy-shen** [comment]
CI should be enabled now. And for pull request, it will not be deployed to https://draw.mogan.app

**Comment by Oyyko** [comment]
Closed. I'll find another way to reduce the size. :)

**Pull request closed without merging**